ops/model.py

```diff
@@ -1053,13 +1068,50 @@ def start(self, *service_names: str):
         """Start given service(s) by name."""
         if not service_names:
             raise TypeError('start expected at least 1 argument, got 0')
-        self._pebble.start_services(service_names)
+
+        if not self.ready:
+            raise PebbleNotReadyError(f"Cannot start {service_names}. Pebble is not ready")
+
+        try:
+            self._pebble.start_services(service_names)
+        except (pebble.APIError, RuntimeError):
+            raise UnknownServiceError(
+                "service(s) {!r} not found. This may be caused by "
+                'querying a service before "add_layer"'.format(service_names)
+            )
+
+    def restart(self, *service_names: str):
```

Comment:
Doesn't stop us from adding this in OF, but note that https://github.com/canonical/pebble/pull/58 (a restart command in Pebble) will likely be merged soon too.

Reply:
Yeh, I feel like this is reasonably safe to add now, with a note to update once the restart API is in place in a version of pebble that's pinned by a `latest/stable` Juju?